src/components/EpisodePreview/EpisodePreview.jsx

```diff
@@ -0,0 +1,143 @@
+import React, { useContext } from "react";
+import {
+	ACTIONS,
+	EpisodesContext as Episodes,
+} from "COMPONENTS/Episodes/";
+import { TrailerContext as Trailer } from "COMPONENTS/Episode/";
+import UTILS from "SHARED/utils.js";
+
+export default function EpisodePreview(props){
+
+	// CONTEXT
+	// ----------------------------
+	const {
+		state: {
+			activeIndex,
+			overlayActive,
+			episodeCount
+		},
+		dispatch
+	} = useContext(Episodes);
+
+
+	const trailers = useContext(Trailer);
+
+	const { src: prevTrailerSrc } = trailers.prev || {};
+	const { src: nextTrailerSrc } = trailers.next || {};
+
+	// RENDER
+	// ----------------------------
+	const { 
+		episodeId,
+		index,
+		number: episodeNo,
+		title,
+		thumbnail: {
+			src: thumbSrc,
+			description: thumbDesc
+		},
+		trailer: {
+			src: trailerSrc
+		}
+	} = props;
+
+	// handle visibility of active episode and episode preview overlay
+	const isActiveEpisode 	= activeIndex === index;
+	const isOverlayActive   = isActiveEpisode && overlayActive;
+	const hidden 			= !isOverlayActive;
+
+	const trailerSrcAttr    = isOverlayActive ? {source: trailerSrc} : {};
+
+	const isFirst 			= index === 0;
+	const isLast  			= index === episodeCount - 1;
+
+	const formattedTitle 	= UTILS.convertToSafeString(title, "-");
+	const previewId 	 	= `${formattedTitle}-preview`;
+
+
+	// CLICK HANDLERS
+	// -----------------------------
+	function setActiveIndex(newIndex, e){
+		e.preventDefault();
+
+		dispatch({
+			type: ACTIONS.GET_EPISODE,
+			value: newIndex
+		});
+	}// changeEpisode
+
+	const prevEpisode  = setActiveIndex.bind(this, index - 1);
+	const nextEpisode  = setActiveIndex.bind(this, index + 1);
```

Comment:
Let's move all these click handlers & binding up above the render section, but also put the two functions together and all the binding together. I get the desire to group variables with their functions, but it's better to group all the functions together in one place than have to go hunt for them.

So the component structure would be:

```
1. HOOKS
2. FUNCTION DECLARATIONS
3. FUNCTION BINDINGS
4. RENDER LOGIC
5. RETURN
```